tm2/pkg/p2p/types.go

```diff
@@ -1,10 +1,115 @@
 package p2p
 
 import (
+	"context"
+	"net"
+
 	"github.com/gnolang/gno/tm2/pkg/p2p/conn"
+	"github.com/gnolang/gno/tm2/pkg/p2p/events"
+	"github.com/gnolang/gno/tm2/pkg/p2p/types"
+	"github.com/gnolang/gno/tm2/pkg/service"
 )
 
 type (
 	ChannelDescriptor = conn.ChannelDescriptor
 	ConnectionStatus  = conn.ConnectionStatus
 )
+
+// Peer is a wrapper for a connected peer
+type Peer interface {
+	service.Service
+
+	FlushStop()
+
+	ID() types.ID         // peer's cryptographic ID
+	RemoteIP() net.IP     // remote IP of the connection
+	RemoteAddr() net.Addr // remote address of the connection
+
+	IsOutbound() bool   // did we dial the peer
+	IsPersistent() bool // do we redial this peer when we disconnect
+	IsPrivate() bool    // do we share the peer
+
+	CloseConn() error // close original connection
+
+	NodeInfo() types.NodeInfo // peer's info
+	Status() ConnectionStatus
+	SocketAddr() *types.NetAddress // actual address of the socket
+
+	Send(byte, []byte) bool
+	TrySend(byte, []byte) bool
```

Comment:
No error handling, so the amount of info you can send to the callee is just yes/no. Also, what is the difference between Send and TrySend?